### PR TITLE
Issue/7152

### PR DESCRIPTION
**Pull request by bluerabbit001**
Issue Number:  7152

Added download progress and cancel features.

### CODE REVIEW

File: ui/component/downloadProgress/view.jsx [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  //     />
+  //   </div>
+  // );
+
+  // console.log('DownloadProgress', props.downloadList);
+  if (!props.downloadList) return null;
+
+  const downloadList = [];
+  Object.keys(props.downloadList).map((key) => {
+    const item = props.downloadList[key];
```
Comment (jessopb) [inline ui/component/downloadProgress/view.jsx:46]: something weird happens here that crashes the app when I play a video.
"cannot read status of null"
redux devtools (which has a bug that sometimes doesn't load unless you `yarn dev` and wait for the app to finish before opening devtools) tells me initial state byOutpoint is { <outpoint>: null } until file_list fetch comes back.

**Review by jessopb — CHANGES_REQUESTED** [review]
The look and feel is great and the code is clean and readable.
Just pointed out some DRY reuse opportunities a couple bugs and codebase style consistency preferences.

**Review by jessopb — CHANGES_REQUESTED** [review]
Changes look better, but:
Card overlay download status styling is invisible on dark mode
Download then Playing another created a condition where a download was stuck - and X-ing the download killed the playing video.

**Comment by jessopb** [comment]
image
Playing a video and then playing another video causes indicator to get stuck at `Infinity minutes NaN`

**Comment by bluerabbit001** [comment]
> I think it might be better to show download progress for all videos downloading whether playing or not. But perhaps disable the 'x' if currently playing - maybe indicate currently playing as well.

It is good idea and I implemented it according to your logic.

**Comment by bluerabbit001** [comment]
> Still getting buggy/stuck display with infinity and NaN when clicking from play to play. Not sure what changed.

clear cache and try again

**Comment by bluerabbit001** [comment]
> Check to see if on your branch, the side menu works while on a player page. It seems to work on master, so might be a rebase issue.

clear cache and try again.

**Comment by jessopb** [comment]
cleared electron application info, cleared the dist folder, checked that most recent branch is `update css` made sure I didn't have other repos linked locally. Your new bouncing notification button shows up. Playing videos do not show up.

**Comment by jessopb** [comment]
closed in favor of rebase

**Pull request closed without merging**